aws/resource_aws_ec2_traffic_mirror_target_test.go

```diff
@@ -0,0 +1,242 @@
+	return fmt.Sprintf(`
+data "aws_availability_zones" "azs" {
+  state = "available"
+}
+
+resource "aws_vpc" "vpc" {
+  cidr_block = "10.0.0.0/16"
+}
+
+resource "aws_subnet" "sub1" {
+  vpc_id = "${aws_vpc.vpc.id}"
+  cidr_block = "10.0.0.0/24"
+  availability_zone = "${data.aws_availability_zones.azs.names[0]}"
+}
+
+resource "aws_subnet" "sub2" {
+  vpc_id = "${aws_vpc.vpc.id}"
+  cidr_block = "10.0.1.0/24"
+  availability_zone = "${data.aws_availability_zones.azs.names[1]}"
+}
+
+resource "aws_lb" "lb" {
+  name               = "%s"
+  internal           = true
+  load_balancer_type = "network"
+  subnets            = ["${aws_subnet.sub1.id}", "${aws_subnet.sub2.id}"]
+
+  enable_deletion_protection  = false
+
+  tags = {
+    Environment = "production"
+  }
+}
+
+resource "aws_ec2_traffic_mirror_target" "target" {
+  description = "%s"
+  network_load_balancer_arn = "${aws_lb.lb.arn}"
+}
+`, lbName, description)
+}
+
+func testAccTrafficMirrorTargetConfigEni(description string) string {
+	return fmt.Sprintf(`
+data "aws_availability_zones" "azs" {
+  state = "available"
+}
+
+data "aws_ami" "amzn-linux" {
+  most_recent = true
+
+  filter {
+    name = "name"
+    values = ["amzn2-ami-hvm-2.0*"]
+  }
+
+  filter {
+    name = "architecture"
+    values = ["x86_64"]
+  }
+
+  owners = ["137112412989"]
+}
+
+resource "aws_vpc" "vpc" {
+  cidr_block = "10.0.0.0/16"
+}
+
+resource "aws_subnet" "sub1" {
+  vpc_id = "${aws_vpc.vpc.id}"
+  cidr_block = "10.0.0.0/24"
+  availability_zone = "${data.aws_availability_zones.azs.names[0]}"
+}
+
+resource "aws_subnet" "sub2" {
+  vpc_id = "${aws_vpc.vpc.id}"
+  cidr_block = "10.0.1.0/24"
+  availability_zone = "${data.aws_availability_zones.azs.names[1]}"
+}
+
+resource "aws_instance" "src" {
+  ami = "${data.aws_ami.amzn-linux.id}"
+  instance_type = "t2.micro"
+  subnet_id = "${aws_subnet.sub1.id}"
+}
+
+resource "aws_ec2_traffic_mirror_target" "target" {
+  description = "%s"
+  network_interface_id = "${aws_instance.src.primary_network_interface_id}"
+}
+`, description)
+}
+
+func testAccPreCheckAWSEc2TrafficMirrorTarget(t *testing.T) {
+	conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+	_, err := conn.DescribeTrafficMirrorTargets(&ec2.DescribeTrafficMirrorTargetsInput{})
+
+	if testAccPreCheckSkipError(err) {
+		t.Skip("skipping traffic mirror target acceprance test: ", err)
+	}
+
+	if err != nil {
+		t.Fatal("Unexpected PreCheck error: ", err)
+	}
+}
+
+func testAccCheckAWSEc2TrafficMirrorTargetDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ec2_traffic_mirror_target" {
+			continue
+		}
+
+		out, err := conn.DescribeTrafficMirrorTargets(&ec2.DescribeTrafficMirrorTargetsInput{
+			TrafficMirrorTargetIds: []*string{
+				aws.String(rs.Primary.ID),
+			},
+		})
+
+		if isAWSErr(err, "InvalidTrafficMirrorTargetId.NotFound", "") {
```

Comment:
Same note here -- if this is the error that is returned when an EC2 Traffic Mirror Target cannot be found, we should include this target resource `Read` functions to trigger Terraform resource recreation:

```go
// In the resource Read functions, potentially in addition to over nil/length checks from DescribeTrafficMirrorTargets call
if isAWSErr(err, "InvalidTrafficMirrorTargetId.NotFound", "") {
	log.Printf("[WARN] EC2 Traffic Mirror Target (%s) not found, removing from state", d.Id())
	d.SetId("")
	return nil
}

if err != nil {
	fmt.Errorf("error describing EC2 Traffic Mirror Target (%s): %w", d.Id(), err)
}
```